### PR TITLE
Provide nodeRef to Transition to fix StrictMode warning

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:
Note: test off the branch locally for strictmode. 

Turn on StrictMode via the checkbox in the storybook top toolbar and test stories that have Popovers/Modals/Trays/ActionBar/any type of overlay. There shouldn't be a strict mode console warning about findDOMNode. Note that there will be a different crash in FocusScope that is fixed by https://github.com/adobe/react-spectrum/pull/3878

Running the strict mode tests will yield a mixed bag of passes/fails due to other strict mode bugs. https://github.com/adobe/react-spectrum/pull/3878 fixes many of the FocusScope related issues

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/overlays/src/Popover.tsx [hunk 1]
```diff
@@ -78,8 +81,9 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
     maxHeight: null
   }, state);
 
+  // Attach Transition's nodeRef to outer most wrapper for node.reflow: https://github.com/reactjs/react-transition-group/blob/c89f807067b32eea6f68fd6c622190d88ced82e2/src/Transition.js#L231
   return (
-    <>
+    <div ref={wrapperRef}>
```
Comment (LFDanLu) [inline packages/@react-spectrum/overlays/src/Popover.tsx:86]: We could instead attach the nodeRef to the Underlay, however it doesn't exist in the `isNonModal` case with popover. Adding a wrapping div should be safe because Underlay and the overlay (Popover, Tray, Modal) all use `position: fixed` or `position: absolute`

**Comment by LFDanLu** [comment]
TODO: rebase

**Comment by LFDanLu** [comment]
@reidbarber the rebase got a bit hairy, seems to have blown away some of your commits. I _think_ I reconstructed the changes you had made, but lemme know if anything was missing

**Comment by LFDanLu** [comment]
Danielbot: https://reactspectrum.blob.core.windows.net/reactspectrum/b04f490826e17e729216611780b71a08c062a444/storybook/index.html?path=/story/accordion--default&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom

File: packages/@react-spectrum/overlays/src/Popover.tsx [hunk 2]
```diff
@@ -78,8 +81,9 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
     maxHeight: null
   }, state);
 
+  // Attach Transition's nodeRef to outer most wrapper for node.reflow: https://github.com/reactjs/react-transition-group/blob/c89f807067b32eea6f68fd6c622190d88ced82e2/src/Transition.js#L231
```
Comment (ktabors) [inline packages/@react-spectrum/overlays/src/Popover.tsx:84]: Why did you add the comment here and not to Modal.tsx or Tray.txs?

**Review by ktabors — COMMENTED** [review]
Code looks good. Haven't tested it yet.

File: packages/@react-spectrum/overlays/src/Popover.tsx (hunk 2 above)
Comment (LFDanLu) [reply packages/@react-spectrum/overlays/src/Popover.tsx:84]: No particular reason, I can definitely add that to those components as well

File: packages/@react-spectrum/overlays/src/Underlay.tsx [hunk 3]
```diff
@@ -11,16 +11,19 @@
...
+import React, {Ref} from 'react';
 import underlayStyles from '@adobe/spectrum-css-temp/components/underlay/vars.css';
 
 interface UnderlayProps {
   isOpen?: boolean,
   isTransparent?: boolean
 }
 
-export function Underlay({isOpen, isTransparent}: UnderlayProps) {
+function Underlay({isOpen, isTransparent}: UnderlayProps, ref: Ref<HTMLDivElement>) {
```
Comment (snowystinger) [inline packages/@react-spectrum/overlays/src/Underlay.tsx:22]: do we still need the ref here? i am not quite seeing where it coming from or being used

Comment (LFDanLu) [reply packages/@react-spectrum/overlays/src/Underlay.tsx:22]: Good catch, forgot to remove that

**Review by snowystinger — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]
LGTM
